src/Qoollo.PerformanceCounters/InternalCounters/Counters/InternalDeltaCounter.cs

```diff
@@ -0,0 +1,178 @@
+            /// Метод создания счётчика из дескриптора
+            /// </summary>
+            /// <returns>Созданный счётчик</returns>
+            public override Counter CreateCounter()
+            {
+                return new InternalNumberOfItemsCounter(this.Name, this.Description);
+            }
+        }
+
+        /// <summary>
+        /// Метод создания дескриптора для InternalNumberOfItemsCounter
+        /// </summary>
+        /// <param name="name">Имя счётчика</param>
+        /// <param name="description">Описание счётчика</param>
+        /// <returns>Дескриптор</returns>
+        internal static InternalCounterDescriptor CreateDescriptor(string name, string description)
+        {
+            return new Descriptor(name, description);
+        }
+
+        // ===================
+
+        /// <summary>
+        /// Конструктор InternalNumberOfItemsCounter
+        /// </summary>
+        /// <param name="name">Имя счётчика</param>
+        /// <param name="description">Описание счётчика</param>
+        public InternalDeltaCounter(string name, string description)
+            : base(name, description)
+        {
+            _currentValue = 0;
+        }
+
+        /// <summary>
+        /// Уменьшить значение на 1
+        /// </summary>
+        /// <returns>Текущее значение или Counter.FailureNum</returns>
+        public override long Decrement()
+        {
+            return Interlocked.Decrement(ref _currentValue);
+        }
+
+        /// <summary>
+        /// Уменьшить значение на "value"
+        /// </summary>
+        /// <param name="value">Значение, на которое уменьшаем</param>
+        /// <returns>Текущее значение или Counter.FailureNum</returns>
+        public override long DecrementBy(long value)
+        {
+            return Interlocked.Add(ref _currentValue, -value);
+        }
+
+        /// <summary>
+        /// Увеличить значение на 1
+        /// </summary>
+        /// <returns>Текущее значение или Counter.FailureNum</returns>
+        public override long Increment()
+        {
+            return Interlocked.Increment(ref _currentValue);
+        }
+
+        /// <summary>
+        /// Увеличить значение на "value"
+        /// </summary>
+        /// <param name="value">Значение, на которое увеличиваем</param>
+        /// <returns>Текущее значение или Counter.FailureNum</returns>
+        public override long IncrementBy(long value)
+        {
+            return Interlocked.Add(ref _currentValue, value);
+        }
+
+        /// <summary>
+        /// Задать значение
+        /// </summary>
+        /// <param name="value">Новое значение</param>
+        public override void SetValue(long value)
+        {
+            Interlocked.Exchange(ref _currentValue, value);
+        }
+
+        /// <summary>
+        /// Получает текущее значение счетчика производительности
+        /// </summary>
+        public override long CurrentValue => Interlocked.Read(ref _currentValue);

```

Comment:
IMO, ```CurrentValue``` should work the same way as ```Measure``` except that it should not update ```_lastMeasuredValue``` field.
```
       public override long CurrentValue 
        { 
            get 
            {
                long currentValue = Interlocked.Read(ref _currentValue);
                long lastMeasuredValue = Interlocked.Read(ref _lastMeasuredValue);
                return currentValue - lastMeasuredValue;
            } 
        }
```

Reply:
In the scenario described below (in my comment to ```Measure()``` method) ```CurrentValue``` can also return negative value. But here it can be fixed simply be reordering reads of ```_currentValue``` and ```_lastMeasuredValue```:

```C#
       public override long CurrentValue 
        { 
            get 
            {
                long lastMeasuredValue = Interlocked.Read(ref _lastMeasuredValue);
                long currentValue = Interlocked.Read(ref _currentValue);
                return currentValue - lastMeasuredValue;
            } 
        }
```

Reply:
Sure, in the case we would have only one value, there would be no need to have the same implementation. Now it is understandable.